proposals/views/attachment_views.py

```diff
@@ -39,11 +39,32 @@ def __init__(self, kind=None, other_object=None, extra=False, **kwargs):
         if not extra:
             del self.fields["kind"]
 
-    def save(
-        self,
-    ):
-        self.instance.kind = self.kind.db_name
-        self.instance.save()
+    def save(self,):
+        # Set the kind if enforced by the view.
+        if self.kind:
+            self.instance.kind = self.kind.db_name
+        # Check we're not editing an attachment that is still in use
+        # by another object.
+        if self.instance.attached_to.count() > 1:
+            # Saving this instance might remove historical data. So
+            # we create a "revision" of sorts. First we save the old
+            # pk:
+            old_pk = self.instance.pk + 0
+            # The following means this instance will get saved under a
+            # new pk, effectively creating a copy.
+            self.instance.pk = None
+            self.instance.id = None
+            self.instance._state.adding = True
+            self.instance.save()
+            # Retrieve and fix the old attachment.
+            instance_manager = type(self.instance).objects
+            old_attachment = instance_manager.get(pk=old_pk)
+            old_attachment.attached_to.remove(self.other_object)
```

Comment:
I was kindoff confused about old_attachment not getting saved here ... But it seems to work as expected, as in, the new object gets removed from the old attachment. I've tested this in the shell.

Reply:
I think elsewhere we do save in these cases "for good measure", but I think that we shouldn't. M2m relationships get updated instantly and it's good to be reminded of that now and again. I also sometimes forget if it's necessary to save them, but it isn't.